src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -205,7 +264,11 @@ public IAsyncCursor<TResult> Execute(IReadBinding binding, CancellationToken can
                 throw new ArgumentException("The binding value passed to ChangeStreamOperation.Execute must implement IReadBindingHandle.", nameof(binding));
             }
 
-            var cursor = Resume(bindingHandle, _resumeAfter, cancellationToken);
+            var cursor = Resume(bindingHandle, cancellationToken);
+            if (_startAtOperationTime == null && _resumeAfter == null)
+            {
+                _startAtOperationTime = binding.Session.OperationTime;
+            }
```

Comment:
This is where we set the startAtOperationTime IFF the user didn't provide either a startAtOperationTime or a resumeAfter value. 

If the user provided either of those values they will be used instead.